test/Fencer/Rules/Test.hs

```diff
@@ -26,20 +26,38 @@ tests = testGroup "Rule tests"
   [ test_rulesLoadRulesYaml
   , test_rulesLoadRulesNonYaml
   , test_rulesLoadRulesRecursively
+  , test_rulesLoadRulesDotDirectory
   ]
 
+-- | A helper function for loading rules and making sure they are as
+-- expected.
+loadRules :: String -> Bool -> IO ()
+loadRules dirTemplate ignoreDotFiles =
+  Temp.withSystemTempDirectory dirTemplate $ \tempDir -> do
+    TIO.writeFile (tempDir </> "config1.yml") domain1Text
+    TIO.writeFile (tempDir </> "config2.yaml") domain2Text
+    definitions <-
+      loadRulesFromDirectory
+        (#directory tempDir)
+        (#ignoreDotFiles ignoreDotFiles)
+    assertEqual "unexpected definitions"
+      (sortOn domainDefinitionId [domain1, domain2])
+      (sortOn domainDefinitionId definitions)
+
+
 -- | test that 'loadRulesFromDirectory' loads rules from YAML files.
 test_rulesLoadRulesYaml :: TestTree
 test_rulesLoadRulesYaml =
   testCase "Rules are loaded from YAML files" $
-    Temp.withSystemTempDirectory "fencer-config" $ \tempDir -> do
-      TIO.writeFile (tempDir </> "config1.yml") domain1Text
-      TIO.writeFile (tempDir </> "config2.yaml") domain2Text
-      definitions <-
-        loadRulesFromDirectory (#directory tempDir) (#ignoreDotFiles True)
-      assertEqual "unexpected definitions"
-        (sortOn domainDefinitionId [domain1, domain2])
-        (sortOn domainDefinitionId definitions)
+    loadRules "fencer-config" True
+
+-- | test that 'loadRulesFromDirectory' loads rules from a
+-- dot-directory when dot-files should be ignored.
+test_rulesLoadRulesDotDirectory :: TestTree
+test_rulesLoadRulesDotDirectory =
+  testCase "Rules are loaded from a dot-directory" $
+    loadRules ".fencer-config" True
```

Comment:
Given our implementation of `loadRulesFromDirectory`, it's possible that it won't apply `#ignoreDotFiles` to the passed directory – only to its subdirectories. So the test should use a different directory structure:

```
<tempdir>
    .subdir
        config.yaml
```

Reply:
I updated this per your comment.